webui/react/src/pages/WorkspaceDetails/WorkspaceQuickSearch.tsx

```diff
@@ -0,0 +1,150 @@
+        const projects = projectResponse.pop();
+        tempWorkspaceMap.set(workspace, projects ?? []);
+      }
+      setWorkspaceMap(tempWorkspaceMap);
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Please try again later.',
+        publicSubject: 'Unable to fetch data.',
+        silent: false,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [ ]);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(e.target.value);
+  };
+
+  const onShowModal = () => {
+    setIsModalVisible(true);
+    fetchData();
+  };
+
+  const onHideModal = () => setIsModalVisible(false);
+
+  const onClickProject = useCallback((project: Project) => {
+    routeToReactUrl(paths.projectDetails(project.id));
+    onHideModal();
+  }, []);
+
+  const onClickWorkspace = useCallback((workspaceId: number) => {
+    routeToReactUrl(paths.workspaceDetails(workspaceId));
+    onHideModal();
+  }, []);
+
+  const treeData: DefaultOptionType[] = useMemo(() => {
+    const data: DefaultOptionType[] = Array.from(workspaceMap)
+      .map(([ workspace, projects ]) => {
+        const treeChildren: DefaultOptionType[] = projects
+          .filter((project) => project.name.includes(searchText))
+          .map((project) => ({
+            title: (
+              <div className={`${css.flexRow} ${css.ellipsis}`}>
+                <Icon name="experiment" />
+                <Link onClick={() => onClickProject(project)}>{project.name}</Link>
+              </div>
+            ),
+            value: `project-${project.id}`,
+          }));
+        return ({
+          children: treeChildren,
+          title: (
+            <div className={`${css.flexRow} ${css.ellipsis}`}>
+              <Icon name="workspaces" />
+              <Link onClick={() => onClickWorkspace(workspace.id)}>{workspace.name}</Link>
+            </div>
+          ),
+          value: `workspace-${workspace.id}`,
+        });
+      })
+      .filter((item) => searchText.length > 0 ? item.children.length > 0 : true);
+    return data;
+  }, [ onClickProject, onClickWorkspace, searchText, workspaceMap ]);
+
+  return (
+    <div>
+      <div onClick={onShowModal}>
+        {children}
+      </div>
+      <Modal
+        closable={false}
+        footer={null}
+        title={(
+          <Input
+            placeholder="Search and Jump to workspace or project"
+            prefix={<Icon name="search" />}
+            width={'100%'}
+            onChange={onChange}
+          />
+        )}
+        visible={isModalVisible}
+        width={'clamp(520px, 50vw, 1000px)'}
+        onCancel={onHideModal}>
+        <div className={css.modalBody}>
+          {isLoading ?
+            <Spinner center tip={'Loading...'} /> : (
+              <>
+                {treeData.length === 0 ?
+                  <Message title="No data found" type={MessageType.Empty} /> : (
```

Comment:
add some flavor? e.g. "No matching projects"

Reply:
Definitely! I'll add `No matching workspace or projects` cuz i fixed the bug